# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: ipc/glue/BackgroundImpl.cpp
# Commit: c8a1e52bfc49
# Full Hash: c8a1e52bfc49cb809bfcdfcd5f3635615579bc86
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2023-11-28 09:49:07
# Description:
#   Bug 1862723 - Use nsCString types for BackgroundParent::KillHardAsync, r=ipc-reviewers,mccr8
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D194548
# ==============================================================================

diff -r 008c188bf70b -r c8a1e52bfc49 ipc/glue/BackgroundImpl.cpp
--- a/ipc/glue/BackgroundImpl.cpp	Mon Nov 27 17:41:45 2023 +0000
+++ b/ipc/glue/BackgroundImpl.cpp	Mon Nov 27 17:49:06 2023 +0000
@@ -186,7 +186,7 @@
 
   // Forwarded from BackgroundParent.
   static void KillHardAsync(PBackgroundParent* aBackgroundActor,
-                            const char* aReason);
+                            const nsACString& aReason);
 
   // Forwarded from BackgroundParent.
   static bool AllocStarter(ContentParent* aContent,
@@ -659,7 +659,7 @@
 
 // static
 void BackgroundParent::KillHardAsync(PBackgroundParent* aBackgroundActor,
-                                     const char* aReason) {
+                                     const nsACString& aReason) {
   ParentImpl::KillHardAsync(aBackgroundActor, aReason);
 }
 
@@ -768,7 +768,7 @@
 
 // static
 void ParentImpl::KillHardAsync(PBackgroundParent* aBackgroundActor,
-                               const char* aReason) {
+                               const nsACString& aReason) {
   AssertIsInMainProcess();
   AssertIsOnBackgroundThread();
   MOZ_ASSERT(aBackgroundActor);
@@ -779,15 +779,16 @@
   MOZ_ASSERT(handle);
 
   MOZ_ALWAYS_SUCCEEDS(NS_DispatchToMainThread(
-      NS_NewRunnableFunction("ParentImpl::KillHardAsync",
-                             [handle = std::move(handle), aReason]() {
-                               mozilla::AssertIsOnMainThread();
+      NS_NewRunnableFunction(
+          "ParentImpl::KillHardAsync",
+          [handle = std::move(handle), reason = nsCString{aReason}]() {
+            mozilla::AssertIsOnMainThread();
 
-                               if (RefPtr<ContentParent> contentParent =
-                                       handle->GetContentParent()) {
-                                 contentParent->KillHard(aReason);
-                               }
-                             }),
+            if (RefPtr<ContentParent> contentParent =
+                    handle->GetContentParent()) {
+              contentParent->KillHard(reason.get());
+            }
+          }),
       NS_DISPATCH_NORMAL));
 
   // After we've scheduled killing of the remote process, also ensure we induce